server/controllers/auth/signup.js

```diff
@@ -0,0 +1,83 @@
+// modules 
+const bcrypt = require('bcrypt');
+const uuid = require('uuid/v1');
+const jwt = require('jsonwebtoken');
```

Comment:
If you plan on providing login access via social media accounts like, Facebook, GitHub, Google, consider using `Passport` instead of `jsonwebtoken`. Because it comes with all those plus jwt.

Reply:
> If you plan on providing login access via social media accounts like, Facebook, GitHub, Google

Well, that's not the plan thought.

But I want the website owner can allow their user to authenticate from their site and re-direct to this site, something like SSO.